### PR TITLE
Filtered out unfriendly error from docker when registry is not reachable.

### LINKED ISSUE #8525: Command line output is unfriendly when getting a 502 from docker
If one gets a 502 response from docker on the image (during e.g. a 'get pods'), we dump the entire error response text to the command line.

### PR DESCRIPTION
**Pull request by dchen1107**
Ran e2e tests, and all passed except ElasticSearch one, but it is marked as flaky for a while.

Fixed #8525

cc/ @vmarmol

### CODE REVIEW

File: pkg/kubelet/dockertools/docker.go [hunk 1]
```diff
@@ -113,6 +115,22 @@ func parseImageName(image string) (string, string) {
...
+	// docker/docker/pull/11314 prints detailed error info for docker pull.
+	// When it hits 502, it returns a verbose html output including an inline svg,
+	// which makes the output of kubectl get pods much harder to parse.
+	// Here converts such verbose output to a concise one.
+	jerr, ok := err.(*jsonmessage.JSONError)
+	if ok && (jerr.Code == http.StatusBadGateway ||
+		jerr.Code == http.StatusServiceUnavailable ||
+		jerr.Code == http.StatusGatewayTimeout) {
+		glog.V(2).Infof("Pulling image %s failed: %v", image, err)
+		return fmt.Errorf("image pull failed for %s because the registry is temporarily unavailbe.", image)
```
Comment (vmarmol) [inline pkg/kubelet/dockertools/docker.go]: nit: prefer %q to %s. It makes it clear what is a variable and helps when the string has whitespace

**Comment by vmarmol** [comment]
LGTM, just a minor nit.

**Comment by dchen1107** [comment]
Done!

**Comment by thockin** [comment]
Did you manually add this to godeps?  It is broken for godep restore - the rev of all the other docker targets is older than this one and they obliterate it.

**Comment by rjnagal** [comment]
@thockin I am going to update all docker pkgs to head. We have half of the docker deps from 1.7-rc branch and others from 1.4. It's stopping dep update for cadvisor.

**Comment by rjnagal** [comment]
btw: 1.4* is the head tag for docker

**Pull request merged**